tests/integration-tests/tests/byos/test_byos.py

```diff
@@ -9,19 +9,172 @@
 
 from tests.common.assertions import assert_head_node_is_running
 
+LAUNCH_TEMPLATES_CONFIG_PATH = "/opt/parallelcluster/shared/launch_templates_config.json"
+BYOS_SUBSTACK_OUTPUTS_PATH = "/opt/parallelcluster/shared/byos_substack_outputs.json"
+BYOS_LOG_PATH = "/var/log/parallelcluster/byos-plugin.log"
+BYOS_HOME = "/home/byos"
+BYOS_USERS_LIST = ["user1", "byosUser"]
+
 
 @pytest.mark.usefixtures("instance", "scheduler", "os")
-def test_byos(region, pcluster_config_reader, clusters_factory):
+def test_byos(region, pcluster_config_reader, s3_bucket_factory, clusters_factory, test_datadir):
     """Test usage of a custom scheduler integration."""
     logging.info("Testing custom scheduler integration.")
 
-    cluster_config = pcluster_config_reader()
+    # Create bucket and upload resources
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    for file in ["byos_infra.cfn.yaml", "artifact"]:
+        bucket.upload_file(str(test_datadir / file), f"byos/{file}")
+    # Create cluster
+    byos_user_list = deepcopy(BYOS_USERS_LIST)
+    cluster_config = pcluster_config_reader(bucket=bucket_name, user1=byos_user_list.pop(), user2=byos_user_list.pop())
     cluster = clusters_factory(cluster_config)
-
+    # Verify head node is running
     assert_head_node_is_running(region, cluster)
+    # Command executor
+    command_executor = RemoteCommandExecutor(cluster)
+    # Start and wait for compute node to setup
+    compute_node = _start_compute_node(region, cluster, command_executor)
+    # Test even handler execution
+    _test_event_handler_execution(command_executor, compute_node)
+    # Test artifacts are downloaded
+    _test_artifacts_download(command_executor)
+    # Test substack outputs
+    _test_subtack_outputs(command_executor)
+    # Test users are created
+    _test_users(command_executor, compute_node)
+
+
+def _get_launch_templates(command_executor):
+    """Get LT from head node"""
+    launch_templates_config_content = command_executor.run_remote_command(f"cat {LAUNCH_TEMPLATES_CONFIG_PATH}").stdout
+    assert_that(launch_templates_config_content).is_not_empty()
+    launch_templates_config = json.loads(launch_templates_config_content)
+    launch_templates_list = []
+    for queue in launch_templates_config.get("Queues").values():
+        for compute_resource in queue.get("ComputeResources").values():
+            launch_templates_list.append(compute_resource.get("LaunchTemplate").get("Id"))
+    assert_that(launch_templates_list).is_length(3)
+
+    return launch_templates_list
+
+
+def _start_compute_node(region, cluster, command_executor):
+    """Start and wait compute node to finish setup"""
+    # Get one launch template for compute node
+    lt = _get_launch_templates(command_executor)[0]
+    run_instance_command = (
+        f"aws ec2 run-instances --region {region}"
+        f" --instance-initiated-shutdown-behavior terminate"
```

Comment:
isn't the shutdown behavior already baked into the LT we provide?

Reply:
it was not, added in https://github.com/aws/aws-parallelcluster/pull/3446